oauth2-client-retrofit2/src/main/java/net/oauth2/client/retrofit/RetrofitTokenService.java

```diff
@@ -160,33 +155,30 @@ public T fetch() throws OAuth2ProtocolException, IOException {
 
 	@SuppressWarnings("unchecked")
 	@Override
-	public T refresh(String refreshToken) throws OAuth2ProtocolException, IOException {
+	public T refresh(String refreshToken) throws IOException {
 		if (refreshToken == null)
 			throw new IllegalArgumentException("refreshToken is null");
 		
 		if (this.refreshTokenGrantRequest == null)
 			throw new IllegalStateException("No refresh token grant initialized. Either authroization server does not support refreshing tokens or fetchToken was never invoked on this instance prior ot invoking refresh.");
 
 		LOGGER.trace("Refreshing Access Token");
-		Map<String, Object> grantRequestFormFrields = null;
+		Map<String, Object> grantRequestFormFields;
 		try {
-			grantRequestFormFrields = this.refreshTokenGrantRequest.map();
+			grantRequestFormFields = this.refreshTokenGrantRequest.map();
 		} catch (Exception e) {
 			throw new IOException(e);
 		}
 
-		Response<String> response = this.tokenService.refreshToken(TokenEndpoint.DEFAULT_URL_PATH, grantRequestFormFrields).execute();
+		Response<String> response = this.tokenService.refreshToken(TokenEndpoint.DEFAULT_URL_PATH, grantRequestFormFields).execute();
 
 		T token = null;
 		if (response.isSuccessful()) {
 			String tokenString = response.body();
 			token = (T) this.objectMapper.readValue(tokenString, this.accessTokenClass);
 			
 			Collection<String> scopes = token.getScopes();
-			String refreshTokenString = token.getRefreshToken();//did we get a new refresh string?
-			if(refreshTokenString == null)
-				refreshTokenString = refreshToken;
```

Comment:
refreshTokenString was being set but the value was not read thereafter, so removed.